plugins/lookup/gcp_secret_manager.py

```diff
@@ -107,103 +132,117 @@
-    from google.oauth2 import service_account
-    from google.auth.transport.requests import AuthorizedSession
-    HAS_GOOGLE_LIBRARIES = True
+    from ansible_collections.google.cloud.plugins.module_utils.gcp_utils import (
+        GcpSession,
+    )
+    HAS_GOOGLE_CLOUD_COLLECTION = True
 except ImportError:
-    HAS_GOOGLE_LIBRARIES = False
+    HAS_GOOGLE_CLOUD_COLLECTION = False
 
-from ansible_collections.google.cloud.plugins.module_utils.gcp_utils import GcpSession, GcpRequest
 from ansible.errors import AnsibleError
+from ansible.utils.display import Display
 
 class GcpLookupException(Exception):
     pass
 
+class GcpMockModule(object):
+    def __init__(self, params):
+        self.params = params
+
+    def fail_json(self, *args, **kwargs):
+        raise AnsibleError(kwargs["msg"])
+
+    def raise_for_status(self, response):
+        try:
+            response.raise_for_status()
+        except getattr(requests.exceptions, "RequestException"):
+            self.fail_json(msg="GCP returned error: %s" % response.json())
+
 class LookupModule(LookupBase):
-    def run(self, terms, variables, **kwargs):
+    def run(self, terms=None, variables=None, **kwargs):
+        self._display = Display()
+        if not HAS_GOOGLE_CLOUD_COLLECTION:
+            raise AnsibleError(
+                "gcp_secret lookup needs a supported version of the google.cloud collection installed. Use `ansible-galaxy collection install google.cloud` to install it"
+        )
         self.set_options(var_options=variables, direct=kwargs)
-        self.scopes = ["https://www.googleapis.com/auth/cloud-platform"]
-        self._validate()
-        self.service_acct_creds = self._credentials()
-        session = AuthorizedSession(self.service_acct_creds)
-        response = session.get("https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{key}/versions/{version}:access".format(**self.get_options()))
-        if response.status_code == 200:
-            result_data = response.json()
-            secret_value = base64.b64decode(result_data['payload']['data'])
-            return [ secret_value ]
+        params = {
+            "key":          self.get_option("key"),
+            "version":      self.get_option("version"),
+            "access_token": self.get_option("access_token"),
+            "scopes":       self.get_option("scopes"),
+            "on_error":     self.get_option("on_error")
+        }
+
+        params['name'] = params['key']
+
+        # support GCP_* env variables for some parameters
+        for param in ["project", "auth_kind", "service_account_file", "service_account_info", "service_account_email", "access_token"]:
+            params[param] = self.fallback_from_env(param)
+
+        self._display.vvv(msg=f"Module Parameters: {params}")
+        fake_module = GcpMockModule(params)
+        result = self.get_secret(fake_module)
+        return [base64.b64decode(result)]
+    
+    def fallback_from_env(self, arg):
```

Comment:
As per above - this is already being handled by `gcp_utils`

Reply:
Once again, thank you for reviewing my code.  Much appreciated.

There is indeed that functionality within the GcpModule class (in the init method for example) [here](https://github.com/ansible-collections/google.cloud/blob/master/plugins/module_utils/gcp_utils.py?plain=1#L282) .  however, in your example, and the one I committed, we don't actually use that class only GcpSession.

I don't believe I can reuse the GcpModule class here in my lookup plugin.  I did actually try this for grins and I got this error when attempting to call my super class constructor:
`TypeError: AnsibleModule.__init__() got multiple values for argument 'argument_spec'`

Not sure this is possible?  What do you think?

And once again, thank you so much.

Reply:
@dcostakos you're right. Sorry, I got mixed up between the module and lookup helper as I just created a PR for a new module.  
Your env handling is absolutely fine. 

Thanks for doing this. Hope your PR gets merged soon, and we can all start using the default GCP collection rather than maintaining separate collections.